spec/core_functions/string/split.hrx

```diff
@@ -0,0 +1,225 @@
+<===> options.yml
+---
+:ignore_for:
+- libsass
+
+<===>
+================================================================================
+<===> empty_string/input.scss
+@use "sass:string";
+a {b: string.split("", "/")}
+
+<===> empty_string/output.css
+a {
+  b: [];
+}
+
+<===>
+================================================================================
+<===> empty_separator/input.scss
+@use "sass:string";
+a {b: string.split("Helvetica", "")}
+
+<===> empty_separator/output.css
+a {
+  b: ["H", "e", "l", "v", "e", "t", "i", "c", "a"];
+}
+
+<===>
+================================================================================
+<===> both_empty/input.scss
+@use "sass:string";
+a {b: string.split("", "")}
+
+<===> both_empty/output.css
+a {
+  b: [];
+}
+
+<===>
+================================================================================
+<===> separator/input.scss
+@use "sass:string";
+a {b: string.split("a, b, c", ", ")}
+
+<===> separator/output.css
+a {
+  b: ["a", "b", "c"];
+}
+
+<===>
+================================================================================
+<===> separator_not_found/input.scss
+@use "sass:string";
+a {b: string.split("a, b, c", "&")}
+
+<===> separator_not_found/output.css
+a {
+  b: ["a, b, c"];
+}
+
+<===>
+================================================================================
+<===> limit/input.scss
+@use "sass:string";
+a {b: string.split("a, b, c, d", ", ", 2)}
+
+<===> limit/output.css
+a {
+  b: ["a", "b", "c, d"];
+}
+
+<===>
+================================================================================
+<===> double_width_character/input.scss
+@use "sass:string";
+// Sass treats strings as sequences of Unicode codepoint; it doesn't care if a
+// character is represented as two UTF-16 code units.
+a {b: string.split("👭a", "")}
+
+<===> double_width_character/output.css
+a {
+  b: ["👭", "a"];
+}
+
+<===>
+================================================================================
+<===> private_use_character/input.scss
+@use "sass:string";
+// Dart Sass emits private-use characters as escapes in expanded mode, but it
+// should still treat them as single characters for the purpose of functions.
+a {b: string.split("\E000", "")}
+
+<===> private_use_character/output.css
+a {
+  b: ["\E000"];
+}
+
+<===>
+================================================================================
+<===> named/input.scss
+@use "sass:string";
+a {b: string.split($string: "a/b/c", $separator: "/", $limit: 1)}
+
+<===> named/output.css
+a {
+  b: ["a", "b/c"];
+}
+
+<===>
+================================================================================
+<===> error/too_many_args/input.scss
+@use "sass:string";
+a {b: string.split("a/b/c", "/", 1, 3)}
+
+<===> error/too_many_args/error
```

Comment:
Note for future reference: if you're just working on tests in isolation, you can leave out the contents of error tests since they can easily be generated from the implementation using `--interactive` mode. The most important thing is _that_ the tests error—the specifics of the message are largely implementation details.